### PR TITLE
Setup/redis mongo

### PR DESCRIPTION
**Pull request by sagar2009kumar**

### CODE REVIEW

File: src/db/mongo.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import mongoose from 'mongoose'
+import { mongoConfig } from '../config'
+
+export const createDBConnection = async () => {
```
Comment (kumaranshu72) [inline src/db/mongo.ts:4]: function value async not required

File: src/db/mongo.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    try {
```
Comment (kumaranshu72) [inline src/db/mongo.ts:5]: try catch not not required here

File: src/db/mongo.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        await mongoose.connect(`${mongoConfig.mongoUrl}`, {
```
Comment (kumaranshu72) [inline src/db/mongo.ts:6]: await is always used in calling funtion return the mongoose.connect here

File: src/server.ts [hunk 4]
```diff
@@ -3,5 +3,5 @@ import app from './app'
 const PORT = process.env.PORT || 3000
 
 app.listen(PORT, () => {
-    console.log(`Magic Happens on port ${PORT}`)
+    global.ctx.log.info(`Magic Happens on port ${PORT}`)
```
Comment (kumaranshu72) [inline src/server.ts:6]: Add console.log also.

Comment (sagar2009kumar) [reply src/server.ts:6]: Why console.log also ?

Comment (kumaranshu72) [reply src/server.ts:6]: to check log while developing on development.

File: src/app.ts [hunk 5]
```diff
@@ -38,6 +41,15 @@ class App {
   private mountRoutes(): void {
     this.app.use(router)
   }
+
+  private async connectDB() {
```
Comment (kumaranshu72) [inline src/app.ts:45]: why this?

Comment (kumaranshu72) [reply src/app.ts:45]: directly call createDBConnection in config()

File: src/db/mongo.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import mongoose from 'mongoose'
+import { mongoConfig } from '../config'
+
+export const createDBConnection = () => {
```
Comment (kumaranshu72) [inline src/db/mongo.ts:4]: export const createDBConnection = () => mongoose.connect(`${mongoConfig.mongoUrl}`, {
        useNewUrlParser: true,
        useCreateIndex: true,
        useUnifiedTopology: true
    })
Change like this

File: src/db/mongo.ts (hunk 1 above)
Comment (sagar2009kumar) [reply src/db/mongo.ts]: Removed the async from here

File: src/db/mongo.ts (hunk 2 above)
Comment (sagar2009kumar) [reply src/db/mongo.ts]: Returned a promise from here.

File: src/db/mongo.ts (hunk 3 above)
Comment (sagar2009kumar) [reply src/db/mongo.ts]: Yes. I have called this in the app.js file.

File: src/app.ts (hunk 5 above)
Comment (sagar2009kumar) [reply src/app.ts:45]: To use the async await, i want the database connection to be ready before the server starts listening. Is it incorrect ?

File: src/db/mongo.ts (hunk 6 above)
Comment (sagar2009kumar) [reply src/db/mongo.ts:4]: Okay. Doing htat.

File: src/app.ts (hunk 5 above)
Comment (kumaranshu72) [reply src/app.ts:45]: No it is not correct. Making DB connection will slow down the server uptime and will lead to performance issues in services which do not require mongo connection.

File: src/db/mongo.ts (hunk 6 above)
Comment (kumaranshu72) [reply src/db/mongo.ts:4]: Can we also rename it to createMongoConnection

File: src/app.ts (hunk 5 above)
Comment (sagar2009kumar) [reply src/app.ts:45]: But suppose in some services, we do need database connection and if the database has not been connected. Will that not affect the availability of the service ?

Comment (kumaranshu72) [reply src/app.ts:45]: Mongoose has inbuild functionality to wait for the query unless db connection is made.

Comment (sagar2009kumar) [reply src/app.ts:45]: Db connection takes time, but what if the database do not get connected at all ? What will happen then ?

Comment (kumaranshu72) [reply src/app.ts:45]: Server will stop if db connection is not established.

File: src/db/mongo.ts (hunk 6 above)
Comment (sagar2009kumar) [reply src/db/mongo.ts:4]: Okay sure.

File: src/app.ts (hunk 5 above)
Comment (sagar2009kumar) [reply src/app.ts:45]: Okay. Let me change it.

Comment (sagar2009kumar) [reply src/app.ts]: Done.

**Pull request merged**